src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -25,7 +25,14 @@ public partial class DependencyObjectCollectionBase : DependencyObject
 	public partial class DependencyObjectCollection<T> : DependencyObjectCollectionBase, IList<T>, IEnumerable<T>, IEnumerable, IObservableVector<T>
 		where T : DependencyObject
 	{
-		public event VectorChangedEventHandler<T> VectorChanged;
+		// Explicit handlers list to avoid the cost of multicast delegates handling
+		private List<VectorChangedEventHandler<T>> _vectorChangedHandlers;
+
+		public event VectorChangedEventHandler<T> VectorChanged
+		{
+			add => (_vectorChangedHandlers ??= new()).Add(value);
+			remove => (_vectorChangedHandlers ??= new()).Remove(value);
+		}
 
```

Comment:
I'm pretty confident that the subscription/unsubscription is implemented in a thread-safe way in WinUI. If you do this:

```csharp
        private async void myButton_Click(object sender, RoutedEventArgs e)
        {
            DependencyObjectCollection x = new();
            for (int i = 0; i < 100; i++)
            {
                int iCopy = i;
                _ = Task.Run(() =>
                {
                    x.VectorChanged += (sender, ev) =>
                    {
                        Debug.WriteLine($"VectorChanged: {iCopy}");
                    };
                });
            }

            await Task.Delay(500); // give a chance to all tasks to be completed.
            x.Add(new Button());
        }
```

This should print the numbers from 0 to 99 (not ordered for sure).

So definitely the change in the PR is a divergence from WinUI. And people sometimes does crazy things with Reactive and execute code in background threads. So it's not very unlikely that this will break something working already.

Reply:
It may be, though in those objects are not supposed to be manipulated from a background thread. Fair comment anyways, it can't be moved away from this kind of behavior.

Reply:
I've adjusted the locking and enumeration, that should do for those scenarios.